control_plane/attachment_service/src/service.rs

```diff
@@ -635,7 +636,7 @@ impl Service {
                 shard_number: tenant_shard_id.shard_number.0 as i32,
                 shard_count: tenant_shard_id.shard_count.0 as i32,
                 shard_stripe_size: create_req.shard_parameters.stripe_size.0 as i32,
-                generation: 0,
+                generation: create_req.generation.map(|g| g as i32).unwrap_or(0),
                 generation_pageserver: i64::MAX,
```

Comment:
Unrelated to this PR, but, follow-up:

Let's have well-defined constants for these sepcial values somewhere.
ALso I'm surprised that generation numbers are i64 and not u64.
Probably because we're going to store it in Postgres?
It has NULLable columns to indicate absence, why not use that.

Reply:
> Probably because we're going to store it in Postgres?

Right.

> It has NULLable columns to indicate absence, why not use that.

Yeah, originally I wanted to avoid dealing with nullability for generations, but in https://github.com/neondatabase/neon/pull/6521 it becomes necessary, so I'll add it there.